dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/SubexpressionPeriodOffsetUtils.java

```diff
@@ -159,9 +162,23 @@ protected static DataQueryParams getParamsWithOffsetPeriods(DataQueryParams para
   // Supportive methods
   // -------------------------------------------------------------------------
 
-  /** Gets the report periods from paramms as a list of Periods. */
+  /**
+   * Gets the report periods from the parameters, or if there are no report periods, gets the filter
+   * periods.
+   *
+   * @param params parameters with reporting periods
+   * @return the report periods
+   */
   private static List<Period> getReportPeriods(DataQueryParams params) {
-    return params.getPeriods().stream().map(Period.class::cast).toList();
+    List<Period> periods = params.getPeriods().stream().map(Period.class::cast).toList();
+
+    if (periods.isEmpty()) {
+      periods = params.getFilterPeriods().stream().map(Period.class::cast).toList();
+    }
+    if (periods.isEmpty()) {
+      throw new IllegalQueryException(ErrorCode.E7150, "No periods found in the parameters.");
```

Comment:
With `ErrorCode`, the idea is that we move the cause/error message to the enum, and avoid text in the exception. So I think this will do here.

```java
throw new IllegalQueryException(ErrorCode.E7150);
```